### DISCUSSION TITLE
Set a static ip for a user ikev2

### DISCUSSION THREAD

**Discussion by Paulo-R-Lima** [strongswan/strongswan#1811] [Q&A]
I have a database that needs to be accessed via vpn, but for that the machine that hosts this database needs to have a static ip, I even managed to define it in ipsec.conf, but despite the connection already having the password saved, when I try connecting the password is required as if it were wrong, but it is not, as soon as I retype it the connection is made and the machine takes the ip defined by me in ipsec.conf, sorry, but I am a new user and I am still learning, thank you for any help! follow my setup!

config setup
    charondebug="ike 1, knl 1, cfg 0"
    uniqueids=no

conn database
    also=ikev2-vpn
    eap_identity=database
    rightsourceip=10.1.0.254

conn Geral
    also=ikev2-vpn
    eap_identity=%identity
    rightsourceip=10.1.0.1-10.1.0.200

conn ikev2-vpn
    auto=add
    compress=no
    type=tunnel
    keyexchange=ikev2
    fragmentation=yes
    forceencaps=yes
    dpdaction=clear
    dpddelay=300s
    rekey=no
    left=%any
    leftid=@xxx.ddns.net
    leftcert=server-cert.pem
    leftsendcert=always
    leftsubnet=0.0.0.0/0
    right=%any
    rightid=%any
    rightauth=eap-mschapv2
    rightsourceip=10.1.0.1-10.1.0.200
    rightdns=8.8.8.8,8.8.4.4
    rightsendcert=never
    eap_identity=%identity
    ike=chacha20poly1305-sha512-curve25519-prfsha512,aes256gcm16-sha384-prfsha384-ecp384,aes256-sha1-modp1024,aes128-sha1-modp1024,3des-sha1-modp1024!
    esp=chacha20poly1305-sha512,aes256gcm16-ecp384,aes256-sha256,aes256-sha1,3des-sha1!

**Comment by tobiasbrunner** [comment]
This is basically a duplicate of #1326.

**Reply by Paulo-R-Lima** [reply]
I apologize, but I'm an inexperienced user, I'm learning now, ok I saw the settings you referenced, I even copied and pasted them, of course making the necessary changes to my environment, but when I connect my native windows client it doesn't pull the settings from swanctl. conf, and yes from ipsec.conf, could you help me? I do not know how to do ! this is my ipesc.conf file.
config setup
    charondebug="ike 1, knl 1, cfg 0"
    uniqueids=yes

conn ikev2-vpn
    auto=add
    compress=no
    type=tunnel
    keyexchange=ikev2
    fragmentation=yes
    forceencaps=yes
    dpdaction=clear
    dpddelay=300s
    rekey=no
    left=%any
    leftid=@vpn.ddns.net
    leftcert=server-cert.pem
    leftsendcert=always
    leftsubnet=0.0.0.0/0
    right=%any
    rightid=%any
    rightsourceip=10.1.0.0/24
    rightauth=eap-mschapv2
    rightdns=8.8.8.8,8.8.4.4
    rightsendcert=never
    eap_identity=%identity
    ike=chacha20poly1305-sha512-curve25519-prfsha512,aes256gcm16-sha384-prfsha384-ecp384,aes256-sha>
    esp=chacha20poly1305-sha512,aes256gcm16-ecp384,aes256-sha256,aes256-sha1,3des-sha1!

**Reply by tobiasbrunner** [reply]
If you use ipsec.conf, follow the instructions in the linked ServerFault answer.

**Reply by Paulo-R-Lima** [reply]
please, see how my ipsec.conf turned out, help me adjust it, see that conn-geral will be a client that will have any ip and conn-Paulo will be a client with a fixed ip, is this correct? if not, where is the error?

config setup
    charondebug="ike 1, knl 1, cfg 0"
    uniqueids=no

conn %default
    auto=add
    compress=no
    type=tunnel
    keyexchange=ikev2
    fragmentation=yes
    forceencaps=yes
    dpdaction=clear
    dpddelay=300s
    rekey=no
    left=%any
    leftid=@vpn.ddns.net
    leftcert=server-cert.pem
    leftsendcert=always
    leftsubnet=0.0.0.0/0
    right=%any
    rightid=%any
    rightauth=eap-mschapv2
    rightsourceip=10.1.0.0/24
    rightdns=8.8.8.8,8.8.4.4
    rightsendcert=never
    ike=chacha20poly1305-sha512-curve25519-prfsha512,aes256gcm16-sha384-prfsha384-ecp384,aes256-sha1-modp1024,aes128-sha1-modp1024,3des-sha1-modp1024!
    esp=chacha20poly1305-sha512,aes256gcm16-ecp384,aes256-sha256,aes256-sha1,3des-sha1!

conn eap-shared
    # Because 'conn %default' has all settings shared between all conn, just:
    auto=ignore

#And the rest is as-is, since the original already has 'auto=add' in each conn:

conn eap-init
   also=eap-shared
   eap_identity=%identity
   rightgroups=%any
   auto=add

conn eap-geral
    also=eap-shared
    eap_identity=%identity
    leftsubnet=10.1.0.0/24

conn Paulo
    also=eap-shared
    eap_identity=Paulo
    leftsubnet=10.1.0.254

**Reply by tobiasbrunner** [reply]
Your connections inherit `auto=ignore` from `eap-shared`, so these won't get loaded, except for `eap-init`, which explicitly sets `auto=add`. Make sure the connections you want to load have `auto=add`. You probably have to list `conn Paulo` before `conn eap-geral` as there won't be a "best" match when switching connections due to the group. Also, if you want to do this via `%default`, you could just remove `eap-shared` and the `also=eap-shared` lines in the connections.
